sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/flow/FlowRuleManager.java

```diff
@@ -61,9 +62,42 @@
     static {
         flowRules.set(Collections.<String, List<FlowRule>>emptyMap());
         currentProperty.addListener(LISTENER);
-        SCHEDULER.scheduleAtFixedRate(new MetricTimerListener(), 0, 1, TimeUnit.SECONDS);
+        startMetricTimerListener();
     }
-
+    
+    /**
+     * <p> Start the MetricTimerListener
+     * <ol>
+     *     <li>if the {@link SentinelConfig#METRIC_FLUSH_INTERVAL} more than 0,
+     * the timer will run with the {@link SentinelConfig#METRIC_FLUSH_INTERVAL} as the rate </li>.
+     *      <li>if the {@link SentinelConfig#METRIC_FLUSH_INTERVAL} less than 0(include) or value is not valid,
+     * then means the timer will not be started </li>
+     * <ol></p>
+     */
+    private static void startMetricTimerListener() {
+        long flushInterval;
+        String flushIntervalStr = SentinelConfig.getConfig(SentinelConfig.METRIC_FLUSH_INTERVAL);
```

Comment:
Can we add a `metricLogFlushIntervalSec()` method in `SentinelConfig` to resolve the metric flushing interval, and leverage the method here?

Reply:
yes, that is more reasonable to put the time valid code into entrance. I will change the code later.